From 2c8924e2ad63f27a677f6366054f61496fbd992d Mon Sep 17 00:00:00 2001
From: Grant Erickson <marathon96@gmail.com>
Date: Sun, 29 Jan 2012 14:18:22 +0200
Subject: [PATCH 08/75] build: Fix install target parent directory
 prerequisites

This changes the install target such that parent directories of
installed paths area created and each path is only installed
on a dependency basis.

Signed-off-by: Grant Erickson <marathon96@gmail.com>
(cherry picked from commit b1117643846bdcd104e04512f84ae131e7e60eb5)
---
 wpa_supplicant/Makefile |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

--- a/wpa_supplicant/Makefile
+++ b/wpa_supplicant/Makefile
@@ -48,9 +48,10 @@ mkconfig:
 	echo CONFIG_DRIVER_HOSTAP=y >> .config
 	echo CONFIG_DRIVER_WEXT=y >> .config
 
-install: all
-	mkdir -p $(DESTDIR)$(BINDIR)
-	for i in $(ALL); do cp $$i $(DESTDIR)$(BINDIR)/$$i; done
+$(DESTDIR)$(BINDIR)/%: %
+	install -D $(<) $(@)
+
+install: $(addprefix $(DESTDIR)$(BINDIR)/,$(ALL))
 	$(MAKE) -C ../src install
 
 OBJS = config.o
